### ISSUE TITLE
Creating truly offline persistence with complementary queries

### ISSUE THREAD

**Issue by karsens** [apollographql/react-apollo#1338]
Hey, first of all. Apollo is awesome :) I'm building a chat in react native and to make it truly offline first I am using redux persist. However, somehow redux-persist seems not to be working that good with the Apollo Redux so I have my own redux store with all chat messages. 

What I'm doing now when loading the messages screen is look at the server for new messages, and look in my own redux store for existing messages that were persisted. 

I figured out a way to do this by creating a sort of Meta-component that uses the Redux State for authorization then sends a query using this authorization and then puts the results of this query into the redux state. 

Unfortunately my solution gives me a warning. It's an anti-pattern!

**TL;DR**: 
I'm basically trying to figure out how to put data from an Apollo query that uses Redux State for Variables into the Redux store without any user-interaction. 
I don't know how to do this right... please help!

**Intended outcome:**
No warnings

**Actual outcome:**
<img width="416" alt="captura de pantalla 2017-11-14 a las 15 05 03" src="https://user-images.githubusercontent.com/1976888/32783798-432f1244-c94d-11e7-93da-daeec995a4e6.png">

**How to reproduce the issue:**
This is my code for the meta component.
ChatScreen is a screen that gets the messages from the redux store. 

```
class _Meta extends React.Component {
  render() {
    const { addMessages, data: { loading, posts, error } } = this.props;

    if (loading || error) {
      return <Text>Loading some messages</Text>; //props to return
    } else {
      addMessages(posts); //make sure posts is an array
      //where to update the redux state? I must be sure to have the data inside of posts... else it won't work.
      return <View />;
    }
  }
}

const Meta = compose(
  connect(({ device }) => ({ device }), disp),
  graphql(POSTS_QUERY, queryConfig)
)(_Meta);

export default compose(
  connect(({ messages }) => ({ messages })),
  createPostMutation
)(ChatScreen);
```

**Version**
- apollo-client@1.X
- react-apollo@1.X

**Comment by hwillson** [comment]
We're cleaning up old issues, and since there hasn't been much traction here, we'll close this off for now. Let us know if you think this issue should stay open (after trying Apollo Client / React Apollo 2.x). Thanks!
